lib/router/index.js

```diff
@@ -106,8 +106,11 @@ export function _rewriteUrlForNextExport (url) {
   path = path.replace(/\/$/, '')
 
   let newPath = path
-  // Append a trailing slash if this path does not have an extension
-  if (!/\.[^/]+\/?$/.test(path)) {
+
+  // By default, append a trailing slash if this path does not have an extension
+  const exportTrailingSlashes = __NEXT_DATA__ &&
+    __NEXT_DATA__.nextExportTrailingSlashes
```

Comment:
So, I was thinking this morning that this might actually be compiled at build time instead of providing it at runtime and having __NEXT_DATA__ used here.

So what you could do is add `DefinePlugin` with `process.env.NEXT_EXPORT_TRAILING_SLASHES` which holds the value of `config.exportTrailingSlashes`. So that it's compiled at build time.

I'm also not sure how you're passing the config to the client right now, so it wouldn't work client-side right 🤔

Reply:
Implemented in #4524